### ISSUE TITLE
Implement downloads via torrents

### ISSUE THREAD

**Issue by f321x** [ollama/ollama#4034]
Model downloads with a slow (10mbit) internet connection are really unreliable and crash around all 5-10gb for me (EOF max retries). 
At the same time huge torrents work very reliable.
If you could implement a call to a external torrent client for model downloading or implement a torrent client the download experience would be more reliable, way faster and you could save costs on hosting the files.

**Comment by bmizerany** [comment]
Hi! We're actively working on fixing issues with regard to slow downloads. We'll continue to improve as we release new versions of Ollama. Thank you for the ticket!

**Comment by TomiWebPro** [comment]
Yes! I signed in just to comment on this one, it would be really great to have all users distribute these models like a torrent network than from a center server which is more costly and slower. I really look forward for this to be implemented on ollama.

**Comment by purificant** [comment]
:+1: for torrents, they can handle large file distribution use cases more gracefully, handling connection drops, pausing, resuming, checksums / verification, etc.
Especially relevant for some of the larger models, for example `llama3.1:405b` is a 231 GB download.

**Comment by lowyek** [comment]
Torrents please!

**Comment by Modzho** [comment]
Yes, we need a torrent option. One disconnect-reconnect and the entire download starts from the beginning. Madness!

**Comment by davidjimenez75** [comment]
Supporting models torrent downloads and USB-based installation would save thousands of terabytes of traffic and energy, and bring Ollama to remote, offline users.

**Comment by ppetr** [comment]
I also up-vote a torrent-based download. With SHA1 or the new SHA256 (BitTorrent v2) it's just as secure, and would be very reliable.

**Comment by panda667** [comment]
Torrents please!!
